### PR TITLE
rebase intl spec onto the current ecma402 master

### PR DESCRIPTION
**Pull request by ryzokuken**
- [x] `dateStyle`/`timeStyle`
- [x] `fractionalSecondDigits`
- [x] `dayPeriod`
- [x] other misc changes

Refs: #1244

### CODE REVIEW

File: spec/intl.html [hunk 1]
```diff
@@ -9,6 +9,9 @@ <h1>Amendments to the ECMAScript® 2021 Internationalization API Specification</
       This section lists amendments which must be made to <a href="https://tc39.es/ecma402/">ECMA-402, the ECMAScript® 2021 Internationalization API Specification</a>.
       Text to be added is marked <ins>like this</ins>, and text to be deleted is marked <del>like this</del>.
     </p>
+    <p>
+      This text is based on top of the ECMA-402 spec text from commit b5edf2cdcca8d69590796cd41abf8399d249b8a1.
```
Comment (Ms2ger) [inline spec/intl.html:13]: Maybe helpful to make the commit hash a link to github

File: spec/intl.html [hunk 2]
```diff
@@ -83,64 +86,6 @@ <h1><a href="https://tc39.es/ecma402/#sec-defaulttimezone">DefaultTimeZone</a> (
   <emu-clause id="sec-datetimeformat-abstracts">
     <h1><a href="https://tc39.es/ecma402/#sec-datetimeformat-abstracts">Abstract Operations For DateTimeFormat Objects</a></h1>
 
-    <emu-table id="table-datetimeformat-components">
```
Comment (Ms2ger) [inline spec/intl.html:86]: add to biblio

File: spec/intl.html [hunk 3]
```diff
@@ -209,15 +154,39 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
...
+        1. Set _dateTimeFormat_.[[DateStyle]] to _dateStyle_.
+        1. Let _timeStyle_ be ? GetOption(_options_, *"timeStyle"*, *"string"*, « *"full"*, *"long"*, *"medium"*, *"short"* », *undefined*).
+        1. Set _dateTimeFormat_.[[TimeStyle]] to _timeStyle_.
+        1. If _dateStyle_ is not *undefined* or _timeStyle_ is not *undefined*, then
+          1. For each row in <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, do
+            1. Let _prop_ be the name given in the Property column of the row.
+            1. Let _p_ be _opt_.[[&lt;_prop_&gt;]].
+            1. If _p_ is not *undefined*, then
+              1. Throw a *TypeError* exception.
+          1. Let _bestFormat_ be DateTimeStyleFormat(_dateStyle_, _timeStyle_, _dataLocaleData_).
```
Comment (Ms2ger) [inline spec/intl.html:174]: Add DateTimeStyleFormat to biblio

File: spec/intl.html (hunk 2 above)
Comment (ryzokuken) [reply spec/intl.html:86]: done

File: spec/intl.html (hunk 3 above)
Comment (ryzokuken) [reply spec/intl.html:174]: done

File: spec/intl.html (hunk 1 above)
Comment (ryzokuken) [reply spec/intl.html:13]: I think the hash text is enough for the case I included it `git diff <hash>` but why not?

Comment (ryzokuken) [reply spec/intl.html:13]: done

**Comment by ryzokuken** [comment]
Ready for review, please take a look.

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**